packages/zendesk-adapter/src/filters/guide_order/guide_orders_utils.ts

```diff
@@ -0,0 +1,155 @@
+*/
+import {
+  BuiltinTypes,
+  Change, CORE_ANNOTATIONS,
+  DeployResult,
+  ElemID, getChangeData,
+  InstanceElement,
+  isReferenceExpression,
+  isRemovalChange, ListType,
+  ObjectType,
+  ReferenceExpression,
+} from '@salto-io/adapter-api'
+import { collections } from '@salto-io/lowerdash'
+import _ from 'lodash'
+import { elements as elementsUtils } from '@salto-io/adapter-components'
+import ZendeskClient from '../../client/client'
+import { API_DEFINITIONS_CONFIG, FilterContext } from '../../config'
+import { BRAND_TYPE_NAME, CATEGORY_TYPE_NAME, SECTION_TYPE_NAME, ZENDESK } from '../../constants'
+import { getZendeskError } from '../../errors'
+
+const { createUrl } = elementsUtils
+const { awu } = collections.asynciterable
+
+export const ORDER_IN_BRAND_TYPE = 'order_in_brand'
+export const ORDER_IN_CATEGORY_TYPE = 'order_in_category'
+export const ORDER_IN_SECTION_TYPE = 'order_in_section'
+
+export const CATEGORIES_FIELD = 'categories'
+export const SECTIONS_FIELD = 'sections'
+export const ARTICLES_FIELD = 'articles'
+
+const GUIDE_ORDER_TYPES : {[key: string]: () => ObjectType} = {
+  [BRAND_TYPE_NAME]: () =>
+    new ObjectType({
+      elemID: new ElemID(ZENDESK, ORDER_IN_BRAND_TYPE),
+      fields: { [CATEGORIES_FIELD]: { refType: new ListType(BuiltinTypes.NUMBER) } },
+    }),
+  [CATEGORY_TYPE_NAME]: () =>
+    new ObjectType({
+      elemID: new ElemID(ZENDESK, ORDER_IN_CATEGORY_TYPE),
+      fields: { [SECTIONS_FIELD]: { refType: new ListType(BuiltinTypes.NUMBER) } },
+    }),
+  [SECTION_TYPE_NAME]: () =>
+    new ObjectType({
+      elemID: new ElemID(ZENDESK, ORDER_IN_SECTION_TYPE),
+      fields: {
+        [SECTIONS_FIELD]: { refType: new ListType(BuiltinTypes.NUMBER) },
+        [ARTICLES_FIELD]: { refType: new ListType(BuiltinTypes.NUMBER) },
+      },
+    }),
+}
+
+export const createOrderType = (typeName: string) : ObjectType => GUIDE_ORDER_TYPES[typeName]()
+
+
+export const createOrderElement = ({ parent, parentField, orderField, childrenElements, orderType }
+: {
+  parent: InstanceElement
+  parentField: string
+  orderField: string
+  childrenElements: InstanceElement[]
+  orderType: ObjectType
+}): InstanceElement => {
+  const parentsChildren = _.orderBy(
+    childrenElements.filter(c => c.value[parentField] === parent.value.id),
+    // Lowest position index first, if there is a tie - the newer is first, another tie - by id
+    ['value.position', 'value.created_at', 'value.id'], ['asc', 'desc', 'asc']
+  )
+
+  return new InstanceElement(
+    `${parent.elemID.name}_${orderField}`,
+    orderType,
+    {
+      [orderField]: parentsChildren.map(c => new ReferenceExpression(c.elemID, c)),
+    },
+    // The same directory as it's parent
+    parent.path && [...parent.path.slice(0, -1), `${orderField}_order`],
+    {
+      [CORE_ANNOTATIONS.PARENT]: [new ReferenceExpression(parent.elemID, parent)],
+    }
+  )
+}
+
+// Transform order changes to new changes and deploy them
```

Comment:
nit:
```suggestion
/**
 * Transform order changes to new changes and deploy them
 */
```
also - `new changes` is not very clear - maybe explain that the changes will be applied on the instances themselves?

Reply:
Old documentation that i forgot to update :/